include/uart.h

```diff
@@ -29,41 +29,218 @@ extern "C" {
 
 #include <device.h>
 
-#ifdef CONFIG_PCI
-#include <drivers/pci/pci.h>
-#include <drivers/pci/pci_mgr.h>
-#endif
+/* @brief Line control signals. */
+enum uart_line_ctrl {
+	UART_LINE_CTRL_RTS = (1 << 1),
+	UART_LINE_CTRL_DTR = (1 << 2),
+	UART_LINE_CTRL_DCD = (1 << 3),
+	UART_LINE_CTRL_DSR = (1 << 4),
+};
+
 /**
- * @brief Options for @a UART initialization.
+ * @brief Types of events passed to callback.
+ *
+ * Following graph represents relations between user function calls and
+ * events received in callback during receiving data on UART
+ *
+ * @note This is for UART_HIGH_LEVEL_API
+ *
+ * rx_en(Buffer 0) (Start receiving)
+ * |
+ * |  rx_buf_rsp(Buffer 1) (Provide second buffer in response to RX_BUF_REQUEST)
+ * |  |
+ * |  |              rx_buf_rsp(Buffer 0) (Buffer 0 can be reused, receiving
+ * |  |              |                     to it has finished)
+ * |  |              |              rx_buf_rsp(Buffer 1)
+ * |  |              |              |
+ * \/ \/             \/             \/
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+->  etc.
```

Comment:
I wanted to make graphical representation as I believe it makes many things easier to understand. I could not find API documentation with graphics to see how it's done.

Reply:
> I wanted to make graphical representation as I believe it makes many things easier to understand.

That can be only appreciated, thanks. Ideally, a fully graphical version would go into the docs later. And well, a picture should help understanding, not replaced textual description.

Reply:
It's even more complex with this conversation in the middle of the diagram.

Reply:
You can uncheck "show comments" option at diff header in github.
Is there any preffered way to include graphics in documentation? @dbkinder

Reply:
ASCII art is a bit of a pain in doxygen.  Because of the processing to generate the HTML documentation, there needs to be at least 4 spaces after the comment asterisk to have the ASCII art treated as a block of text with a fixed-width font, (otherwise it will ignore white space and format it as a paragraph).  If the ASCII art is overly complex, you can also use a doxygen @image command:   ``@image html myimage.jpg`` for example, and we'd need to tweak the doxygen configuration (IMAGE_PATH) to identify where the image folder lives (we wouldn't want to put ``myimage.jpg`` in the ``/include`` folder).  A third alternative is to refer to a documentation page where the jpg file is shown, or we also support using graphviz in the documentation (using .dot files).

Reply:
@Mierunski: I don't remember if I made that suggestion explicitly or not, but IMHO a drawn diagram should be an appendix, not replacement of textual description. Plus, docstrings are just docstrings, they aren't replacement of a real API overview documentation (as living in doc/).

So, on the diagram, time goes from left to right, in a linear manner. Why not write a list like:

1. You call X1.
2. You receive even X2.
3. You call X3.
4. You receive X4.
...

Once you to that, I may imagine you'll find it possible to remove asciiart diagram *from the doctring* completely, and instead a bit later find time to draw a proper .png for the overview docs (and link it from docstring if needed).

Just my 2 cents anyway.